packages/react-query-4/src/SuspenseInfiniteQuery.test-d.tsx

```diff
@@ -15,7 +15,15 @@ describe('<SuspenseInfiniteQuery/>', () => {
         //@ts-expect-error no suspense
         suspense={boolean}
       >
-        {(query) => <>{query.data.pages.filter(({ text }) => text)}</>}
+        {(query) => (
+          <>
+            {query.data.pages
+              .filter(({ text }) => text)
+              .map((item, index) => (
+                <div key={index}>{item.text}</div>
+              ))}
+          </>
+        )}
```

Comment:
Let me explain these changes:

After upgrading to the latest typescript and typescript-eslint, I got this error from ci:type

<img width="546" alt="Captura de pantalla 2024-11-28 a las 11 19 55 p  m" src="https://github.com/user-attachments/assets/a41ebd0f-edcc-48df-8fc6-3aecb72fca2e">

In order to have `ReactNode`, I added `<div>` element instead of just having `text`. 
This fixed the typescript error